src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2257,5 +2257,100 @@ private static bool TryFindFirstMatchedLane(Vector128<byte> mask, Vector128<byte
             matchedLane = BitOperations.TrailingZeroCount(selectedLanes) >> 2;
             return true;
         }
+
+        public static void Reverse(ref byte buf, nuint length)
+        {
+            int numBytes = (int)length;
+            ref byte first = ref buf;
+            ref byte last = ref Unsafe.Subtract(ref Unsafe.Add(ref first, numBytes), 1);
+            int numBytesWritten = 0;
+            if (Avx2.IsSupported && Vector256<byte>.Count * 2 <= numBytes)
+            {
+                Vector256<byte> reverseMask = Vector256.Create(
+                    (byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, // first 128-bit lane
+                    15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0); // first 128-bit lane
+                last = ref Unsafe.Subtract(ref Unsafe.Add(ref first, numBytes), Vector256<byte>.Count);
+                do
+                {
+                    // Load in values from beginning and end of the array.
+                    Vector256<byte> tempFirst = Unsafe.As<byte, Vector256<byte>>(ref first);
+                    Vector256<byte> tempLast = Unsafe.As<byte, Vector256<byte>>(ref last);
```

Comment:
nit:
```suggestion
                    Vector256<byte> tempFirst = Vector256<byte>.LoadUnsafe(ref first);
                    Vector256<byte> tempLast = Vector256<byte>.LoadUnsafe(ref last);
```

Reply:
-- These are new APIs since the PR was started, they help cleanup exactly this scenario